catalog/cataloger_get_entry.go

```diff
@@ -55,8 +57,37 @@ func (c *cataloger) GetEntryMaybeExpired(ctx context.Context, repository, refere
 	return res.(*Entry), nil
 }
 
+func (c *cataloger) GetEntryMaybeExpiredBatched(ctx context.Context, repository, reference string, path string) (*Entry, error) {
+	if err := Validate(ValidateFields{
+		{Name: "repository", IsValid: ValidateRepositoryName(repository)},
+		{Name: "reference", IsValid: ValidateReference(reference)},
+	}); err != nil {
+		return nil, err
+	}
+	if path == "" {
+		return nil, db.ErrNotFound
+	}
+	ref, err := ParseRef(reference)
+	if err != nil {
+		return nil, err
+	}
+	entry, err := c.dbBatchEntryRead(repository, path, *ref)
```

Comment:
Can we enforce a timeout and/or cancellation from `ctx` here?  I think we should!
Otherwise suppose something is down and blocks a single read out of each batch... and a Spark job starts.  We have a huge number of pending reads at this point.  Every read times out on the client, and _the client will retry a few times_ -- so even more pending reads.  And when it unblocks we get a thundering herd here.
It would happen without batching too, batching just makes it worse.

Reply:
Not sure I understand. And I think batching makes the herd smaller :-)